application/contract/rootdomain/rootdomain.go

```diff
@@ -30,139 +29,131 @@ import (
+	res := map[string]interface{}{
+		"rootDomain":             rd.GetReference().String(),
+		"rootMember":             rd.RootMember.String(),
+		"migrationDaemonMembers": migrationDaemonsMembersOut,
+		"migrationAdminMember":   rd.MigrationAdminMember.String(),
+		"nodeDomain":             rd.NodeDomain.String(),
+	}
+	resJSON, err := json.Marshal(res)
 	if err != nil {
-		return map[string]interface{}{"member": name},
-			fmt.Errorf("[ getUserInfoMap ] Can't get total balance: %s", err.Error())
+		return nil, fmt.Errorf("failed to marshal result: %s", err.Error())
 	}
-	return map[string]interface{}{
-		"member": name,
-		"wallet": balance,
-	}, nil
+	return resJSON, nil
 }
 
-// DumpUserInfo processes dump user info request
-func (rd *RootDomain) DumpUserInfo(reference string) ([]byte, error) {
-	caller := *rd.GetContext().Caller
-	ref, err := insolar.NewReferenceFromBase58(reference)
-	if err != nil {
-		return nil, fmt.Errorf("[ DumpUserInfo ] Failed to parse reference: %s", err.Error())
-	}
-	if *ref != caller && caller != rd.RootMember {
-		return nil, fmt.Errorf("[ DumpUserInfo ] You can dump only yourself")
-	}
-	m := member.GetObject(*ref)
+func (rd *RootDomain) AddBurnAddresses(burnAddresses []string) error {
+	rd.FreeBurnAddresses = append(rd.FreeBurnAddresses, burnAddresses...)
 
-	res, err := rd.getUserInfoMap(m)
-	if err != nil {
-		return nil, fmt.Errorf("[ DumpUserInfo ] Problem with making request: %s", err.Error())
-	}
+	return nil
+}
 
-	return json.Marshal(res)
+func (rd *RootDomain) AddBurnAddress(burnAddress string) error {
+	rd.FreeBurnAddresses = append(rd.FreeBurnAddresses, burnAddress)
+
+	return nil
 }
 
-// DumpAllUsers processes dump all users request
-func (rd *RootDomain) DumpAllUsers() ([]byte, error) {
-	if *rd.GetContext().Caller != rd.RootMember {
-		return nil, fmt.Errorf("[ DumpAllUsers ] Only root can call this method")
-	}
-	res := []map[string]interface{}{}
-	iterator, err := rd.NewChildrenTypedIterator(member.GetPrototype())
-	if err != nil {
-		return nil, fmt.Errorf("[ DumpAllUsers ] Can't get children: %s", err.Error())
+func (rd RootDomain) GetBurnAddress() (string, error) {
+	if len(rd.FreeBurnAddresses) == 0 {
+		return "", fmt.Errorf("no more burn address left")
 	}
 
-	for iterator.HasNext() {
-		cref, err := iterator.Next()
-		if err != nil {
-			return nil, fmt.Errorf("[ DumpAllUsers ] Can't get next child: %s", err.Error())
-		}
-
-		if cref == rd.RootMember {
-			continue
-		}
-		m := member.GetObject(cref)
-		userInfo, _ := rd.getUserInfoMap(m)
-		// XXX: we ignore error here as some users may miss wallet for now
-		if userInfo != nil {
-			res = append(res, userInfo)
-		}
-	}
-	resJSON, _ := json.Marshal(res)
-	return resJSON, nil
+	return rd.FreeBurnAddresses[0], nil
 }
 
-var INSATTR_Info_API = true
+func (rd *RootDomain) AddNewMemberToMaps(publicKey string, burnAddress string, memberRef insolar.Reference) error {
+	rd.PublicKeyMap[TrimPublicKey(publicKey)] = memberRef
```

Comment:
You didn't check if pk already exists in map - is rewriting member ref in that case ok?